src/types.rs

```diff
@@ -14,12 +14,31 @@
 // Software.
 
 #![allow(unused_assignments)]
+extern crate sodiumoxide;
+
+use sodiumoxide::crypto;
 
 use cbor::CborTagEncode;
 use rustc_serialize::{Decodable, Decoder, Encodable, Encoder};
 
+pub fn vector_as_u8_64_array(vector: Vec<u8>) -> [u8;64] {
```

Comment:
Yes I think it should as it's simpler and the compiler does not need to bounds check, so its also more efficient.